src/tests/dbus-tests/test_vdo.py

```diff
@@ -0,0 +1,284 @@
+import os
+import re
+import time
+import six
+import dbus
+import unittest
+
+import udiskstestcase
+
+
+class UdisksVDOTest(udiskstestcase.UdisksTestCase):
+    '''This is a basic VDO test suite'''
+
+    LOOP_DEVICE_PATH = '/var/tmp/udisks_test_disk_vdo'
+
+    @classmethod
+    def setUpClass(cls):
+        udiskstestcase.UdisksTestCase.setUpClass()
+        if not cls.check_module_loaded('VDO'):
+            udiskstestcase.UdisksTestCase.tearDownClass()
+            raise unittest.SkipTest('Udisks module for VDO tests not loaded, skipping.')
+
+    def setUp(self):
+        # create backing sparse file
+        # VDO needs at least 5G of space and we need some room for the grow test
+        # ...rumors go that vdo internally operates on 2G extents...
+        self.run_command('truncate -s 8G %s' % self.LOOP_DEVICE_PATH)
+        ret_code, self.dev_name = self.run_command('losetup --find --show %s' % self.LOOP_DEVICE_PATH)
+        self.assertEqual(ret_code, 0)
+        time.sleep(0.5)
+        self.device = self.get_device(self.dev_name)
+        self.assertIsNotNone(self.device)
+        super().setUp()
+
+    def tearDown(self):
+        # tear down loop device
+        self.run_command('losetup --detach %s' % self.dev_name)
+        os.remove(self.LOOP_DEVICE_PATH)
+        super().tearDown()
+
+    def _force_remove(self, vdo_name):
+         if os.path.exists('/dev/mapper/%s' % vdo_name):
+             ret, _out = self.run_command('vdo stop --force --name %s' % vdo_name)
+             if not ret == 0:
+                 self.fail('Failed to stop the vdo volume %s' % vdo_name)
+             ret, _out = self.run_command('vdo remove --force --name %s' % vdo_name)
+             if not ret == 0:
+                 self.fail('Failed to remove vdo volume %s' % vdo_name)
+
+
+
```

Comment:
Extra blank lines here.

Reply:
That was on purpose but I'll remove it :-)